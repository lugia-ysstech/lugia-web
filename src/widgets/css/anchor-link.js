/**
 * Anchor
 * create by guorg
 * @flow
 */
import { px2emcss } from '../css/units';
import styled from 'styled-components';
import get from './theme-common-dict';

export type AnchorLinkProps = {
  title: string | React.ReactNode,
  href: string,
  children?: any,
  onClick?: Function,
  active?: boolean,
};
export type AnchorLinkState = {};
type CSSProps = {
  active?: boolean,
};

const FontSize = 1.2;
const em = px2emcss(FontSize);

export const LinkWrap = styled.div`
  font-size: ${FontSize}rem;
  padding-top: ${em(12)};
  padding-left: ${em(get('marginToSameElement'))};
  line-height: 1;
`;

const getLinkColor = (props: CSSProps) => {
  const { active } = props;
  let color = 'darkGreyColor';
  if (active) {
    color = 'themeColor';
  }
  return `color: ${get(color)};`;
};

export const Link = styled.a`
  display: block;
  ${getLinkColor};
  transition: all 0.3s ease-in-out;
  text-decoration: none;
  &:hover {
    color: ${get('themeColor')};
  }
  &:focus {
    text-decoration: none;
  }
`;
