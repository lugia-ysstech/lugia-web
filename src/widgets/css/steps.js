import { px2emcss } from './units';
import type { ThemeType } from '@lugia/lugia-web';
export type StepType = 'simple' | 'flat' | 'icon' | 'dot';
export type StepStatus = 'finish' | 'process' | 'next' | 'wait' | 'error';
export type SizeType = 'normal' | 'mini';
export type OrientationType = 'horizontal' | 'vertical';
export type AlignType = 'center' | 'left';
const em = px2emcss(1.2);

const lightThemeColor = '$lugia-dict.@lugia/lugia-web.themeHoverColor';
const themeColor = '$lugia-dict.@lugia/lugia-web.themeColor';
const successColor = '$lugia-dict.@lugia/lugia-web.successColor';
const dangerColor = '$lugia-dict.@lugia/lugia-web.dangerColor';
const lightGreyColor = '$lugia-dict.@lugia/lugia-web.lightGreyColor';
const defaultColor = '$lugia-dict.@lugia/lugia-web.defaultColor';

export const getStepColor = (props: Object) => {
  const { stepStatus, stepType } = props;
  let color = defaultColor;
  switch (stepStatus) {
    case 'finish':
      if (stepType === 'simple' || stepType === 'dot') return (color = successColor);
      if (stepType === 'flat') return (color = lightThemeColor);
      return (color = themeColor);
    case 'process':
      if (stepType === 'flat') return (color = lightThemeColor);
    case 'next':
      color = themeColor;
      break;
    case 'wait':
      color = lightGreyColor;
      break;
    case 'error':
      color = dangerColor;
      break;
    default:
      break;
  }
  return color;
};
export const getStepBackgroundColor = (props: Object) => {
  const { stepStatus, stepType } = props;
  const color =
    (stepStatus === 'finish' || stepStatus === 'process') && stepType === 'flat'
      ? lightThemeColor
      : stepStatus === 'process'
      ? themeColor
      : stepStatus === 'error' && stepType === 'flat'
      ? dangerColor
      : defaultColor;
  return color;
};
export const getIndex = props => {
  const { stepType, stepStatus } = props;
  const index = stepType === 'flat' && (stepStatus === 'wait' || stepStatus === 'next') ? 9 : 11;
  return `z-index: ${index};`;
};
export const getWidth = props => {
  const { theme, orientation } = props;
  const { width } = theme;
  const theWidth = orientation === 'vertical' ? '' : em(width) ? em(width) : em(500);
  return ` width: ${theWidth};`;
};
