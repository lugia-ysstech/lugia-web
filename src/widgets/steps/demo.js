/**
 *
 * create by liangguodong on 2018/10/22
 *
 * @flow
 */
import * as React from 'react';
import Steps from './steps';
import Step from './step';
import Button from '../button/index';
import Widget from '../consts/index';
import Theme from '../theme/';
import styled from 'styled-components';

const Wrapper = styled.div`
  text-align: left;
  margin: 50px;
`;
const VWrapper = styled.div`
  text-align: left;
  margin: 50px;
  display: inline-block;
`;

const steps = [
  {
    title: 'First Step',
    description: 'First Step Content',
  },
  {
    title: 'Second Step',
    description: 'Second  Step Content',
  },
  {
    title: 'Last Step',
    description: 'Third  Step Content',
  },
];
const StepDescription = styled.div`
  width: 400px;
  margin-top: 40px;
  border: 1px solid #ccc;
  height: 100px;
  text-align: center;
  padding-top: 20px;
`;

class StepsDemo extends React.Component<Object, Object> {
  constructor(props) {
    super(props);
    this.state = {
      currentStepNumber: 1,
    };
  }
  next() {
    const currentStepNumber =
      this.state.currentStepNumber > 3 ? 1 : this.state.currentStepNumber + 1;
    this.setState({ currentStepNumber });
  }
  pre() {
    const currentStepNumber =
      this.state.currentStepNumber <= 1 ? 3 : this.state.currentStepNumber - 1;
    this.setState({ currentStepNumber });
  }
  render() {
    const { currentStepNumber } = this.state;
    const matchCurrentNumber = currentStepNumber > 0 && currentStepNumber <= 3;
    const theCurrentStepNumber = matchCurrentNumber ? currentStepNumber : 1;
    const description = matchCurrentNumber ? steps[theCurrentStepNumber - 1].description : '';
    const view = {
      [Widget.Steps]: {
        StepsOutContainer: {
          normal: {
            width: 400,
          },
        },
      },
      [Widget.Button]: {
        Container: {
          normal: {
            width: 80,
            margin: {
              top: 10,
              left: 10,
            },
          },
        },
      },
    };
    return (
      <Wrapper>
        <Theme config={view}>
          <Steps currentStepNumber={currentStepNumber} stepType={'simple'} size={'normal'}>
            {steps.map((item, i) => (
              <Step currentStepNumber={currentStepNumber} title={item.title} />
            ))}
          </Steps>
          <StepDescription>{description}</StepDescription>
          <Button type="primary" onClick={() => this.pre()}>
            pre
          </Button>
          <Button type="primary" onClick={() => this.next()}>
            Next
          </Button>
        </Theme>
      </Wrapper>
    );
  }
}

const hThemeConfig = {
  [Widget.Steps]: {
    StepsOutContainer: {
      normal: {
        width: 800,
      },
    },
  },
};
const vThemeConfig = {
  [Widget.Steps]: {
    StepsOutContainer: {
      normal: {
        margin: 10,
      },
    },
  },
};
const singleThemeConfig = {
  [Widget.Step]: {
    StepLine: {
      normal: {
        background: {
          color: 'purple',
        },
      },
    },
  },
};
const data = [
  { title: 'title1111', description: 'description111', stepStatus: 'finish' },
  { title: 'title222', description: 'description2', stepStatus: 'process' },
  { title: 'title33', description: 'description33', stepStatus: 'next' },
  { title: 'title444', description: 'description444', stepStatus: 'wait' },
  { title: 'title555', description: 'description5555', stepStatus: 'error' },
];
export default () => {
  return (
    <div>
      <Wrapper>
        <Steps data={data} />
      </Wrapper>
    </div>
  );
};
