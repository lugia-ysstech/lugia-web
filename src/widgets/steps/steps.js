/**
 *
 * create by liangguodong on 2018/10/22
 *
 * @flow
 */
import '../common/shirm';
import React, { Component } from 'react';
import Widget from '../consts/index';

import KeyBoardEventAdaptor from '../common/KeyBoardEventAdaptor';
import ThemeProvider from '../theme-provider';
import type { AlignType, StepType, OrientationType, SizeType } from '../css/steps';
import Step from './step';
import { isHorizontal } from './step';
import { getAttributeFromObject } from '../common/ObjectUtils';
import CSSComponent, { css, StaticComponent } from '@lugia/theme-css-hoc';
import { deepMerge } from '@lugia/object-utils';

const StepsOutContainer = CSSComponent({
  tag: 'div',
  className: 'StepsOutContainer',
  normal: {
    selectNames: [['width'], ['height'], ['margin'], ['padding'], ['background'], ['opacity']],
    getThemeMeta(themeMeta, themeProps) {
      const { width, height } = themeMeta;
      const { propsConfig } = themeProps;
      const { orientation } = propsConfig;
      const theWidth = width && width > 0 ? width : '';
      const theHeight = height && height > 0 ? height : '';
      return isHorizontal(orientation) ? { width: theWidth } : { height: theHeight };
    },
  },
  css: css`
    display: inline-block;
    position: relative;
    font-size: 1.2rem;
  `,
});

const HStepsOutContainer = CSSComponent({
  tag: 'div',
  className: 'HStepsOutContainer',
  normal: {
    selectNames: [['width'], ['height']],
    getCSS(themeMeta, themeProps) {
      const { propsConfig } = themeProps;
      const { orientation } = propsConfig;
      const direction = isHorizontal(orientation) ? 'row' : 'column';
      return ` flex-direction: ${direction};`;
    },
    getThemeMeta(themeMeta, themeProps) {
      const { propsConfig } = themeProps;
      const { orientation } = propsConfig;
      return isHorizontal(orientation) ? { width: '100%' } : { height: '100%' };
    },
  },
  css: css`
    font-variant: tabular-nums;
    box-sizing: border-box;
    list-style: none;
    display: flex;
  `,
});
const ContentContainer = CSSComponent({
  tag: 'div',
  className: 'StepsContentContainer',
  normal: {
    selectNames: [['width'], ['height']],
    getCSS(themeMeta: Object, themeProps: Object) {
      const { propsConfig } = themeProps;
      const { orientation } = propsConfig;
      const display = isHorizontal(orientation) ? 'block' : 'inline-block';
      return `display:${display};`;
    },
  },
});
const OrientationContainer = StaticComponent({
  tag: 'div',
  className: 'StepsContentContainer',
  css: css`
    display: ${props => (isHorizontal(props.orientation) ? 'block' : 'inline-block')};
  `,
});

type StepsState = {
  _renderAgain: boolean,
};

type StepsProps = {
  stepType: StepType,
  currentStepNumber: number,
  orientation: OrientationType,
  size: SizeType,
  getTheme: Function,
  children: React$Element<any>,
  desAlign: AlignType,
  data: Array<Object>,
  defaultData: Array<Object>,
  themeProps: Object,
  getPartOfThemeProps: Function,
  getPartOfThemeHocProps: Function,
};
export const defaultData = [
  { title: 'step1', stepStatus: 'finish' },
  {
    title: 'step2',
    stepStatus: 'process',
  },
  {
    title: 'step3',
    stepStatus: 'next',
  },
  { title: 'step4', stepStatus: 'wait' },
];

class Steps extends Component<StepsProps, StepsState> {
  static defaultProps = {
    currentStepNumber: 0,
    stepType: 'simple',
    size: 'normal',
    orientation: 'horizontal',
    desAlign: 'left',
    defaultData,
  };
  static displayName = Widget.Steps;
  state = { _renderAgain: false };
  constructor(props: StepsProps) {
    super(props);
  }

  static getDerivedStateFromProps(props: StepsProps, state: StepsState) {}

  render() {
    const { orientation } = this.props;
    const theThemeProps = this.props.getPartOfThemeProps('StepsOutContainer', {
      props: {
        orientation,
      },
    });
    const normalConfig = isHorizontal(orientation)
      ? { height: this.contentHeight }
      : { width: this.contentWidth };
    const contentThemeProps = deepMerge(
      { themeConfig: { normal: normalConfig } },
      this.props.getPartOfThemeProps('StepsContentContainer', {
        props: {
          orientation,
        },
      })
    );
    return (
      <StepsOutContainer themeProps={theThemeProps} orientation={orientation}>
        <div>
          <OrientationContainer>{this.getHSteps()}</OrientationContainer>
          <ContentContainer themeProps={contentThemeProps} />
        </div>
      </StepsOutContainer>
    );
  }

  getHSteps() {
    const { orientation, stepType } = this.props;
    const theThemeProps = this.props.getPartOfThemeProps('StepsContainer', {
      props: {
        orientation,
      },
    });
    return (
      <HStepsOutContainer orientation={orientation} stepType={stepType} themeProps={theThemeProps}>
        {this.getChildren()}
      </HStepsOutContainer>
    );
  }
  getStepsConfig(child: Object, i: number) {
    const {
      orientation,
      stepType,
      size,
      currentStepNumber,
      desAlign,
      getPartOfThemeHocProps,
    } = this.props;
    return {
      ...getPartOfThemeHocProps('Step'),
      orientation,
      stepType,
      size,
      stepNumber: i + 1,
      isFirst: i === 0,
      currentStepNumber,
      desAlign,
      title: getAttributeFromObject(
        child,
        'title',
        getAttributeFromObject(child.props, 'title', '')
      ),
      stepStatus: getAttributeFromObject(
        child,
        'stepStatus',
        getAttributeFromObject(child.props, 'stepStatus', undefined)
      ),
      description: getAttributeFromObject(
        child,
        'description',
        getAttributeFromObject(child.props, 'description', '')
      ),
      icon: getAttributeFromObject(
        child,
        'icon',
        getAttributeFromObject(child.props, 'icon', 'lugia-icon-financial_cloud')
      ),
      isDashed: getAttributeFromObject(
        child,
        'isDashed',
        getAttributeFromObject(child.props, 'isDashed', false)
      ),
      getChildWidths: this.getChildWidths,
    };
  }
  childrenLength = 0;

  getChildren() {
    const { children, data, defaultData } = this.props;
    const finalData = data
      ? this.data2Step(data)
      : Array.isArray(children) && children.length > 0
      ? React.Children.map(children, (child, i) => {
          return React.cloneElement(child, this.getStepsConfig(child, i));
        })
      : this.data2Step(defaultData);
    this.childrenLength = finalData.length;
    return finalData;
  }

  getMaxNumber(array: Array<number>) {
    return Math.max.apply(null, array);
  }

  titleWidthArray = [];
  titleHeightArray = [];
  descWidthArray = [];
  descHeightArray = [];
  childArray = [];
  contentWidth = 0;
  contentHeight = 0;
  getChildWidths = (obj: Object) => {
    const { titleWidth, titleHeight, descWidth, descHeight } = obj;
    this.titleWidthArray.push(titleWidth);
    this.titleHeightArray.push(titleHeight);
    this.descWidthArray.push(descWidth);
    this.descHeightArray.push(descHeight);

    if (this.childrenLength === this.titleWidthArray.length) {
      const maxDescWidth = this.getMaxNumber(this.descWidthArray);
      const maxDescHeight = this.getMaxNumber(this.descHeightArray);
      const maxTitleWidth = this.getMaxNumber(this.titleWidthArray);
      const maxTitleHeight = this.getMaxNumber(this.titleHeightArray);
      this.contentWidth =
        maxDescWidth && maxDescWidth > maxTitleWidth ? maxDescWidth : maxTitleWidth;
      this.contentHeight = maxDescHeight ? maxTitleHeight + maxDescHeight : maxTitleHeight;
    }
    this.setState({ _renderAgain: true });
  };

  data2Step(data: Array<Object>) {
    return data.map((child, i) => {
      return this.getStep(child, i);
    });
  }

  getStep(child: Object, i: number) {
    return <Step {...this.getStepsConfig(child, i)} />;
  }
}

const TargetSteps = ThemeProvider(KeyBoardEventAdaptor(Steps), Widget.Steps);
export default TargetSteps;
