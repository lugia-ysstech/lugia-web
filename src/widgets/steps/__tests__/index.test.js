/**
 *
 * create by lianggd
 *
 * @flow
 */
import React from 'react';
import renderer from 'react-test-renderer';
import Wrapper from '../demo';
import { _Step } from '../step';
import 'jest-styled-components';
import { getStepColor, getStepBackgroundColor } from '../../css/steps';
import type { StepStatus } from '../../css/steps';
const themeColor = '$lugia-dict.@lugia/lugia-web.themeColor';
const themeHoverColor = '$lugia-dict.@lugia/lugia-web.themeHoverColor';
const successColor = '$lugia-dict.@lugia/lugia-web.successColor';
const dangerColor = '$lugia-dict.@lugia/lugia-web.dangerColor';
const defaultColor = '$lugia-dict.@lugia/lugia-web.defaultColor';

describe('stepsDemo', () => {
  it('Component JSON', () => {
    const renders = renderer.create(<Wrapper />);
    expect(renders.toJSON()).toMatchSnapshot();
  });

  function TestgetStepValue(stepNumber: number, status: StepStatus, exceptNumber: number) {
    it('getIconByStepStatus', () => {
      expect(_Step.prototype.getStepValue(stepNumber, status)).toBe(exceptNumber);
    });
  }
  TestgetStepValue(3, 'finish', 3);
  TestgetStepValue(2, 'process', 2);
  TestgetStepValue(1, 'wait', 0);
  TestgetStepValue(0, 'process', 0);
  TestgetStepValue(1, 'process', 1);
  function TestgetIcon(status: StepStatus, exceptIcon: string) {
    it('getIconByStepStatus', () => {
      expect(_Step.prototype.getIconByStepStatus(status)).toBe(exceptIcon);
    });
  }
  TestgetIcon('finish', 'lugia-icon-reminder_check');
  TestgetIcon('process', '');
  TestgetIcon('next', '');
  TestgetIcon('error', 'lugia-icon-reminder_close');

  function testGetStepColor(props: Object, exceptColor) {
    it('getStepColor', () => {
      expect(getStepColor(props)).toBe(exceptColor);
    });
  }
  testGetStepColor({ stepStatus: 'finish', stepType: 'simple' }, successColor);
  testGetStepColor({ stepStatus: 'finish', stepType: 'flat' }, themeHoverColor);
  testGetStepColor({ stepStatus: 'finish', stepType: 'dot' }, successColor);
  testGetStepColor({ stepStatus: 'process', stepType: 'flat' }, themeHoverColor);
  testGetStepColor({ stepStatus: 'next', stepType: 'simple' }, themeColor);
  testGetStepColor({ stepStatus: 'error', stepType: 'simple' }, dangerColor);
  function testGetStepBackgroundColor(props: Object, exceptColor) {
    it('getStepBackgroundColor', () => {
      expect(getStepBackgroundColor(props)).toBe(exceptColor);
    });
  }
  testGetStepBackgroundColor({ stepStatus: 'finish', stepType: 'simple' }, defaultColor);
  testGetStepBackgroundColor({ stepStatus: 'finish', stepType: 'flat' }, themeHoverColor);
  testGetStepBackgroundColor({ stepStatus: 'process', stepType: 'flat' }, themeHoverColor);
  testGetStepBackgroundColor({ stepStatus: 'next', stepType: 'simple' }, defaultColor);
  testGetStepBackgroundColor({ stepStatus: 'error', stepType: 'flat' }, dangerColor);
  testGetStepBackgroundColor({ stepStatus: 'error', stepType: 'simple' }, defaultColor);
});
