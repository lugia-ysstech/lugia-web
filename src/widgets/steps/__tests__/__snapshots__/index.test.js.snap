// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stepsDemo Component JSON 1`] = `
.c7 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  display: inline-block;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  text-align: center;
  font-weight: 700;
  line-height: 1;
  z-index: 5;
}

.c10 {
  position: relative;
  cursor: pointer;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c8 {
  position: absolute;
  white-space: normal;
  display: block;
  overflow: hidden;
  word-break: keep-all;
  font-size: 1.4rem;
  margin-top: 3rem;
  text-align: left;
}

.c9 {
  position: absolute;
  white-space: normal;
  display: block;
  overflow: hidden;
  word-break: keep-all;
  font-size: 1.2rem;
  text-align: left;
  bottom: -4rem;
}

.c11 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  position: relative;
  z-index: 10;
  padding: 0 0.6rem;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c12 {
  position: relative;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
}

.c3 {
  position: relative;
}

.c13 {
  display: inLine-block;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  text-align: center;
  font-weight: 700;
  line-height: 1;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  border-radius: 50%;
}

.c4 {
  height: 100%;
  width: 100%;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c5 {
  text-align: center;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  z-index: 11;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

.c1 {
  display: inline-block;
  position: relative;
  font-size: 1.2rem;
}

.c2 {
  font-variant: tabular-nums;
  box-sizing: border-box;
  list-style: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c0 {
  text-align: left;
  margin: 50px;
}

<div>
  <div
    className="c0"
  >
    <div
      className="c1"
      orientation="horizontal"
      style={Object {}}
    >
      <div
        className="c2"
        orientation="horizontal"
        style={
          Object {
            "width": "100%",
          }
        }
      >
        <div
          className="c3"
          style={
            Object {
              "width": "3.2rem",
            }
          }
        >
          <div
            className="c4"
            style={Object {}}
          >
            <div
              className="c5"
              style={
                Object {
                  "height": "3.2rem",
                  "width": "3.2rem",
                }
              }
            >
              <div
                className="c6"
                orientation="horizontal"
                size="normal"
                style={
                  Object {
                    "backgroundColor": "#ffffff",
                    "borderBottomColor": "#56c22d",
                    "borderBottomStyle": "solid",
                    "borderBottomWidth": "0.1rem",
                    "borderLeftColor": "#56c22d",
                    "borderLeftStyle": "solid",
                    "borderLeftWidth": "0.1rem",
                    "borderRightColor": "#56c22d",
                    "borderRightStyle": "solid",
                    "borderRightWidth": "0.1rem",
                    "borderTopColor": "#56c22d",
                    "borderTopStyle": "solid",
                    "borderTopWidth": "0.1rem",
                    "height": "3.2rem",
                    "width": "3.2rem",
                  }
                }
              >
                <i
                  className="lugia-icon-reminder_check   c7"
                  onClick={[Function]}
                  onMouseDown={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  style={
                    Object {
                      "color": "#56c22d",
                      "fontSize": "1.8rem",
                    }
                  }
                />
              </div>
              <div
                className="c8"
                style={
                  Object {
                    "color": "#333333",
                  }
                }
              >
                title1111
                 
              </div>
              <div
                className="c9"
                style={
                  Object {
                    "color": "#666666",
                  }
                }
              >
                description111
              </div>
            </div>
          </div>
        </div>
        <div
          className="c10"
          style={
            Object {
              "width": "21.2rem",
            }
          }
        >
          <div
            className="c3"
            style={
              Object {
                "width": "100%",
              }
            }
          >
            <div
              className="c4"
              style={Object {}}
            >
              <div
                className="c11"
                style={
                  Object {
                    "width": "100%",
                  }
                }
              >
                <div
                  className="c12"
                  style={
                    Object {
                      "backgroundColor": "#4d63ff",
                      "height": "0.1rem",
                      "width": "100%",
                    }
                  }
                />
              </div>
              <div
                className="c5"
                style={
                  Object {
                    "height": "3.2rem",
                    "width": "3.2rem",
                  }
                }
              >
                <div
                  className="c6"
                  orientation="horizontal"
                  size="normal"
                  style={
                    Object {
                      "backgroundColor": "#4d63ff",
                      "borderBottomColor": "#4d63ff",
                      "borderBottomStyle": "solid",
                      "borderBottomWidth": "0.1rem",
                      "borderLeftColor": "#4d63ff",
                      "borderLeftStyle": "solid",
                      "borderLeftWidth": "0.1rem",
                      "borderRightColor": "#4d63ff",
                      "borderRightStyle": "solid",
                      "borderRightWidth": "0.1rem",
                      "borderTopColor": "#4d63ff",
                      "borderTopStyle": "solid",
                      "borderTopWidth": "0.1rem",
                      "height": "3.2rem",
                      "width": "3.2rem",
                    }
                  }
                >
                  <i
                    className="   c7"
                    onClick={[Function]}
                    onMouseDown={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    style={
                      Object {
                        "color": "#4d63ff",
                        "fontSize": "1.8rem",
                      }
                    }
                  />
                  <div
                    className="c13"
                    size="normal"
                    style={
                      Object {
                        "color": "#ffffff",
                        "fontSize": "2rem",
                        "fontWeight": 700,
                      }
                    }
                  >
                    2
                  </div>
                </div>
                <div
                  className="c8"
                  style={
                    Object {
                      "color": "#333333",
                    }
                  }
                >
                  title222
                   
                </div>
                <div
                  className="c9"
                  style={
                    Object {
                      "color": "#666666",
                    }
                  }
                >
                  description2
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="c10"
          style={
            Object {
              "width": "21.2rem",
            }
          }
        >
          <div
            className="c3"
            style={
              Object {
                "width": "100%",
              }
            }
          >
            <div
              className="c4"
              style={Object {}}
            >
              <div
                className="c11"
                style={
                  Object {
                    "width": "100%",
                  }
                }
              >
                <div
                  className="c12"
                  style={
                    Object {
                      "backgroundColor": "#4d63ff",
                      "height": "0.1rem",
                      "width": "100%",
                    }
                  }
                />
              </div>
              <div
                className="c5"
                style={
                  Object {
                    "height": "2rem",
                    "width": "2rem",
                  }
                }
              >
                <div
                  className="c6"
                  orientation="horizontal"
                  size="normal"
                  style={
                    Object {
                      "backgroundColor": "#ffffff",
                      "borderBottomColor": "#4d63ff",
                      "borderBottomStyle": "solid",
                      "borderBottomWidth": "0.1rem",
                      "borderLeftColor": "#4d63ff",
                      "borderLeftStyle": "solid",
                      "borderLeftWidth": "0.1rem",
                      "borderRightColor": "#4d63ff",
                      "borderRightStyle": "solid",
                      "borderRightWidth": "0.1rem",
                      "borderTopColor": "#4d63ff",
                      "borderTopStyle": "solid",
                      "borderTopWidth": "0.1rem",
                      "height": "2rem",
                      "width": "2rem",
                    }
                  }
                >
                  <i
                    className="   c7"
                    onClick={[Function]}
                    onMouseDown={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    style={
                      Object {
                        "color": "#4d63ff",
                        "fontSize": "1.8rem",
                      }
                    }
                  />
                  <div
                    className="c13"
                    size="normal"
                    style={
                      Object {
                        "color": "#ffffff",
                        "fontSize": "2rem",
                        "fontWeight": 700,
                      }
                    }
                  >
                    0
                  </div>
                </div>
                <div
                  className="c8"
                  style={
                    Object {
                      "color": "#cccccc",
                    }
                  }
                >
                  title33
                   
                </div>
                <div
                  className="c9"
                  style={
                    Object {
                      "color": "#cccccc",
                    }
                  }
                >
                  description33
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="c10"
          style={
            Object {
              "width": "21.2rem",
            }
          }
        >
          <div
            className="c3"
            style={
              Object {
                "width": "100%",
              }
            }
          >
            <div
              className="c4"
              style={Object {}}
            >
              <div
                className="c11"
                style={
                  Object {
                    "width": "100%",
                  }
                }
              >
                <div
                  className="c12"
                  style={
                    Object {
                      "backgroundColor": "#cccccc",
                      "height": "0.1rem",
                      "width": "100%",
                    }
                  }
                />
              </div>
              <div
                className="c5"
                style={
                  Object {
                    "height": "2rem",
                    "width": "2rem",
                  }
                }
              >
                <div
                  className="c6"
                  orientation="horizontal"
                  size="normal"
                  style={
                    Object {
                      "backgroundColor": "#ffffff",
                      "borderBottomColor": "#cccccc",
                      "borderBottomStyle": "solid",
                      "borderBottomWidth": "0.1rem",
                      "borderLeftColor": "#cccccc",
                      "borderLeftStyle": "solid",
                      "borderLeftWidth": "0.1rem",
                      "borderRightColor": "#cccccc",
                      "borderRightStyle": "solid",
                      "borderRightWidth": "0.1rem",
                      "borderTopColor": "#cccccc",
                      "borderTopStyle": "solid",
                      "borderTopWidth": "0.1rem",
                      "height": "2rem",
                      "width": "2rem",
                    }
                  }
                >
                  <i
                    className="   c7"
                    onClick={[Function]}
                    onMouseDown={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    style={
                      Object {
                        "color": "#4d63ff",
                        "fontSize": "1.8rem",
                      }
                    }
                  />
                  <div
                    className="c13"
                    size="normal"
                    style={
                      Object {
                        "color": "#ffffff",
                        "fontSize": "2rem",
                        "fontWeight": 700,
                      }
                    }
                  >
                    0
                  </div>
                </div>
                <div
                  className="c8"
                  style={
                    Object {
                      "color": "#cccccc",
                    }
                  }
                >
                  title444
                   
                </div>
                <div
                  className="c9"
                  style={
                    Object {
                      "color": "#cccccc",
                    }
                  }
                >
                  description444
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="c10"
          style={
            Object {
              "width": "21.2rem",
            }
          }
        >
          <div
            className="c3"
            style={
              Object {
                "width": "100%",
              }
            }
          >
            <div
              className="c4"
              style={Object {}}
            >
              <div
                className="c11"
                style={
                  Object {
                    "width": "100%",
                  }
                }
              >
                <div
                  className="c12"
                  style={
                    Object {
                      "backgroundColor": "#f22735",
                      "height": "0.1rem",
                      "width": "100%",
                    }
                  }
                />
              </div>
              <div
                className="c5"
                style={
                  Object {
                    "height": "3.2rem",
                    "width": "3.2rem",
                  }
                }
              >
                <div
                  className="c6"
                  orientation="horizontal"
                  size="normal"
                  style={
                    Object {
                      "backgroundColor": "#ffffff",
                      "borderBottomColor": "#f22735",
                      "borderBottomStyle": "solid",
                      "borderBottomWidth": "0.1rem",
                      "borderLeftColor": "#f22735",
                      "borderLeftStyle": "solid",
                      "borderLeftWidth": "0.1rem",
                      "borderRightColor": "#f22735",
                      "borderRightStyle": "solid",
                      "borderRightWidth": "0.1rem",
                      "borderTopColor": "#f22735",
                      "borderTopStyle": "solid",
                      "borderTopWidth": "0.1rem",
                      "height": "3.2rem",
                      "width": "3.2rem",
                    }
                  }
                >
                  <i
                    className="lugia-icon-reminder_close   c7"
                    onClick={[Function]}
                    onMouseDown={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    style={
                      Object {
                        "color": "#f22735",
                        "fontSize": "1.8rem",
                      }
                    }
                  />
                </div>
                <div
                  className="c8"
                  style={
                    Object {
                      "color": "#f22735",
                    }
                  }
                >
                  title555
                   
                </div>
                <div
                  className="c9"
                  style={
                    Object {
                      "color": "#f22735",
                    }
                  }
                >
                  description5555
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
